### PR TITLE
adds support for loading parameters in /tokens endpoint

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- adds support for loading parameters in /tokens endpoint

## Why are we making these changes?

We would like to avoid an N+1 query pattern to perform a search on tokens and then another set of token retrievals to load information of specific fields.

### CODE REVIEW

File: waiter/src/waiter/token.clj [hunk 1]
```diff
@@ -668,6 +690,12 @@
                           (string? owner-param) #{owner-param}
                           (coll? owner-param) (set owner-param)
                           :else (list-token-owners kv-store))
+                 include-param (get request-params "include")
```
Comment (kevo1ution) [inline waiter/src/waiter/token.clj:693]: It seems that the `include` query parameter will have several meanings. It is used for `metadata`, `deleted`, `service-id`, `cluster`; maybe we can create a new query field name instead? Does `fields` sound okay? so:
`/tokens?fields=metadata` will provide token `{"metadata": {}, "token":"kitchen", ...other default index fields`, and still includes the default fields.

Here is an example `fields` interface: https://developers.facebook.com/docs/graph-api/overview#fields

Comment (kevo1ution) [reply waiter/src/waiter/token.clj:693]: The response will not have a nested `parameters` response anymore

Comment (shamsimam) [reply waiter/src/waiter/token.clj]: We spoke offline and decided to use the `parameters` query parameter. We did not go with a flat map due to the conflict with the `maintenance` field already displayed in the `/tokens` output today.

File: waiter/src/waiter/token.clj [hunk 2]
```diff
@@ -646,6 +646,27 @@
...
+  (->> parameters
+    (set)
+    (map #(-> % (str/split #"\.") (vec)))
+    (filter #(contains? sd/token-data-keys (first %)))
+    (vec)))
+
+(defn extract-token-data-param-vals
+  "Extracts the values of parameters from the provided token description."
+  [token-description display-params-keys]
+  (loop [[entry-keys & remaining-keys] display-params-keys
```
Comment (kevo1ution) [inline waiter/src/waiter/token.clj:661]: nitpick: This can be a `reduce`

**Review by kevo1ution — DISMISSED** [review]

File: waiter/src/waiter/token.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/src/waiter/token.clj]: Changed `loop` to `reduce`.

**Review by kevo1ution — APPROVED** [review]

**Pull request merged**